packages/in_app_purchase/in_app_purchase_storekit/darwin/in_app_purchase_storekit/Sources/in_app_purchase_storekit/StoreKit2/InAppPurchasePlugin+StoreKit2.swift

```diff
@@ -146,6 +145,31 @@ extension InAppPurchasePlugin: InAppPurchase2API {
     }
   }
 
+  /// Wrapper method around StoreKit2's countryCode() method
+  /// https://developer.apple.com/documentation/storekit/storefront/countrycode
+  func countryCode(completion: @escaping (Result<String, Error>) -> Void) {
+    Task {
+      guard let currentStorefront = await Storefront.current else {
+        let error = PigeonError(
+          code: "storekit2_failed_to_fetch_country_code",
+          message: "Storekit has failed to fetch the country code.",
+          details: "Storekit has failed to fetch the country code.")
```

Comment:
is `details` optional? can we pass nil?

Reply:
its not optional

Reply:
it should be optional (see the code gen https://github.com/flutter/packages/pull/8900/files#diff-3f480a81033c1685aa12f185e9e6196630d77a183c0d6fc39ecd19b8170bed20R21)